### ISSUE TITLE
Producing the JSON samples from XML via metaschema conversions

### ISSUE THREAD

**Issue by wendellpiez** [usnistgov/OSCAL#241]
# User Story:

We still do not have a complete and comprehensive spec for the XML<->JSON mapping (which is subject to finalization of the metaschema and OSCAL declarations models), but we have demo code that shows such conversions live via scripts produced from a metaschema; these can presumably be applied to make equivalent JSON from any (valid) XML we make or acquire.

To promote consistency and comprehensibility, JSON samples in the repository should be produced by these conversion scripts not by other means.

## Goals:

Place the metaschema-supported XML-JSON conversion technology "under load". Produce consistent JSON examples easily. Avoid having to maintain two sets of examples in parallel by hand.

## Dependencies:

- [x] Determine scope of coverage for samples conversion (maybe start small)
- [x] Test XSLTs and shake out the results of the conversions themselves and make sure they hang together
- [x] Agree on acceptable stack/tools dependencies (Java/SaxonHE from command line, i.e. same as metaschema build?)

## Acceptance Criteria

Demonstrate that we can produce the JSON examples via automated means that can be (a) distributed and shared to support this task and (b) capable of being used under CI.

Integrating this capability fully under CI is not an acceptance criterion.

**Comment by wendellpiez** [comment]
NB by "JSON examples" here we mean not only the examples given in metaschema docs (see #242) but all the provided JSON variants of published catalogs and profiles (in the `content` directory). Ultimately this script may run broadly over subdirectories of content.

Also - an XSLT 3.0 could do this instead of a shell script, picking up sets of XML and dynamically applying conversions to them per detected type or location.

**Comment by wendellpiez** [comment]
We are now producing JSON successfully by automated conversion inside and outside CI/CD. What is missing is the specification for how Metaschema constructs map into XML and JSON. This will also be a basis for more formal work (see #239, #388).

This spec can be in markdown to start. (It is tempting to make a metaschema for the Metaschema but the models will not go: it would be only a shell for documentation.) The ToC should also be compared to query returns from the Metaschema XSD to ensure comprehensiveness.

However, this documentation is not listed as an Acceptance Criterion for this Issue, which is now being met. I suggest we either edit the statement of work or close this Issue and open one for writing a specification. (Unless there are also other work items here: @david-waltermire-nist any thoughts?)

**Comment by wendellpiez** [comment]
A new issue #398 describes a requirement for mapping paths in conversion between XML and JSON. The mechanism we build needs to be written up with the work here.

**Comment by wendellpiez** [comment]
@david-waltermire-nist @anweiss you might be in the best position to assess whether:

1. JSON samples are being successfully produced in CI/CD?
2. Can we close this Issue in favor of other issues focusing on requirements?

Once I am also comfortable running the build scripts I could also vouch for this.

**Comment by david-waltermire-nist** [comment]
This has been completed.
